src/exception/exception-manifest/util.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { DeepReadonly } from 'deep-freeze';
+import { Donor, FollowUp, Specimen, Treatment } from '../../clinical/clinical-entities';
+import {
+	ExceptionType,
+	ProgramPropertyExceptionRecord,
+	EntityPropertyExceptionRecord,
+} from './types';
+import {
+	EntityExceptionRecord,
+	ProgramExceptionRecord,
+	isSpecimenExceptionRecord,
+	isTreatmentExceptionRecord,
+	isFollowupExceptionRecord,
+} from '../property-exceptions/types';
+
+export const getSpecimenId = (
```

Comment:
These 3 functions could likely be combined into a single generic function, writing this way just the path of least resistance

Reply:
Updated to use a generic function, see https://github.com/icgc-argo/argo-clinical/pull/1153/commits/fcc95d8a79ceb0c0d78d868d82470f24a2597b80 to compare with the previous version